### PR TITLE
Flyover text added for open choice view

### LINKED ISSUE #1516: Modal, Open choice view does not show fly-over text.
**Describe the bug**
Modal and open choice view does not show fly over text on inclusion of flyover code to questionnaire-itemControl extension.

e.g 
"extension": [
            {
              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
              "valueCodeableConcept": {
                "coding": [
                  {
                    "system": "http://hl7.org/fhir/questionnaire-item-control",
                    "code": "flyover",
                    "display": "Fly-over"
                  }
                ],
                "text": "Flyover"
              }
            }
          ]

**Component**
SDC library

**To Reproduce**

**Expected behavior**
flyover text should be shown.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: [e.g. Pixel4a emulator]
 - Android version: [e.g. Settings -> About phone -> Android version]
 - Build number: [e.g. Settings -> About phone -> Build number]

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1516

**Description**
1. added support for flyover text in in open choice view in SDC library
2. Updated json for open choice component with flyover extension
3. Updated json for Modal component with flyover extension

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix 
**Screenshots (if applicable)**
FlyoverForOpenChoiceView

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
@PallaviGanorkar 
Can you please also attached screenshot for `Modal component` with flyover text?

**Comment by PallaviGanorkar** [comment]
> @PallaviGanorkar Can you please also attach screenshot for `Modal component` with flyover text?

I have already attached screen from catalog app open choice component, Are you expecting something else?

File: catalog/src/main/assets/modal-questionnaire.json [hunk 1]
```diff
@@ -6,6 +6,28 @@
...
+          "type": "display",
+          "extension": [
+            {
+              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
+              "valueCodeableConcept": {
+                "coding": [
+                  {
+                    "system": "http://hl7.org/fhir/questionnaire-item-control",
+                    "code": "flyover",
+                    "display": "Fly-over"
```
Comment (santosh-pingle) [inline catalog/src/main/assets/modal-questionnaire.json:22]: nit : let's remove line no. 22.

File: catalog/src/main/assets/modal-questionnaire.json [hunk 2]
```diff
@@ -6,6 +6,28 @@
...
+                  }
+                ],
+                "text": "Flyover"
```
Comment (santosh-pingle) [inline catalog/src/main/assets/modal-questionnaire.json:25]: nit : let's remove line no. 25.

File: catalog/src/main/assets/modal_with_validation_questionnaire.json [hunk 3]
```diff
@@ -7,6 +7,28 @@
...
+          "type": "display",
+          "extension": [
+            {
+              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
+              "valueCodeableConcept": {
+                "coding": [
+                  {
+                    "system": "http://hl7.org/fhir/questionnaire-item-control",
+                    "code": "flyover",
+                    "display": "Fly-over"
```
Comment (santosh-pingle) [inline catalog/src/main/assets/modal_with_validation_questionnaire.json:23]: nit: ditto

File: catalog/src/main/assets/modal_with_validation_questionnaire.json [hunk 4]
```diff
@@ -7,6 +7,28 @@
...
+                  }
+                ],
+                "text": "Flyover"
```
Comment (santosh-pingle) [inline catalog/src/main/assets/modal_with_validation_questionnaire.json:26]: nit: ditto.

File: catalog/src/main/assets/open-choice-questionnaire.json [hunk 5]
```diff
@@ -19,8 +20,29 @@
...
+          "type": "display",
+          "extension": [
+            {
+              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
+              "valueCodeableConcept": {
+                "coding": [
+                  {
+                    "system": "http://hl7.org/fhir/questionnaire-item-control",
+                    "code": "flyover",
+                    "display": "Fly-over"
```
Comment (santosh-pingle) [inline catalog/src/main/assets/open-choice-questionnaire.json:37]: nit: ditto

File: catalog/src/main/assets/open-choice-questionnaire.json [hunk 6]
```diff
@@ -19,8 +20,29 @@
...
+                  }
+                ],
+                "text": "Flyover"
```
Comment (santosh-pingle) [inline catalog/src/main/assets/open-choice-questionnaire.json:40]: nit : ditto

File: catalog/src/main/assets/open_choice_with_validation_questionnaire.json [hunk 7]
```diff
@@ -22,6 +22,28 @@
...
+          "type": "display",
+          "extension": [
+            {
+              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-itemControl",
+              "valueCodeableConcept": {
+                "coding": [
+                  {
+                    "system": "http://hl7.org/fhir/questionnaire-item-control",
+                    "code": "flyover",
+                    "display": "Fly-over"
```
Comment (santosh-pingle) [inline catalog/src/main/assets/open_choice_with_validation_questionnaire.json:38]: nit : ditto

File: catalog/src/main/assets/open_choice_with_validation_questionnaire.json [hunk 8]
```diff
@@ -22,6 +22,28 @@
...
+                  }
+                ],
+                "text": "Flyover"
```
Comment (santosh-pingle) [inline catalog/src/main/assets/open_choice_with_validation_questionnaire.json:41]: nit: ditto

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDialogSelectViewHolderFactory.kt [hunk 9]
```diff
@@ -57,6 +58,8 @@ internal object QuestionnaireItemDialogSelectViewHolderFactory :
 
       override fun bind(questionnaireItemViewItem: QuestionnaireItemViewItem) {
         cleanupOldState()
+        holder.summaryHolder.hint =
```
Comment (santosh-pingle) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDialogSelectViewHolderFactory.kt:61]: If possible can you please check whether unit test can be written to assert whether text is assigned to hint field?

**Review by santosh-pingle — COMMENTED** [review]
It looks good to me, except few optional comments. thanks!

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDialogMultiSelectViewHolderFactoryEspressoTest.kt [hunk 10]
```diff
@@ -180,6 +186,46 @@ class QuestionnaireItemDialogMultiSelectViewHolderFactoryEspressoTest {
     assertThat(questionnaireItemViewItem.answers).isEmpty()
   }
 
+  @Test
+  fun bindView_SetHintText() {
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDialogMultiSelectViewHolderFactoryEspressoTest.kt:190]:
```suggestion
  fun bindView_setHintText() {
```

File: catalog/src/main/assets/open_choice_with_validation_questionnaire.json (hunk 8 above)
Comment (jingtang10) [reply catalog/src/main/assets/open_choice_with_validation_questionnaire.json:39]: this is not done?

File: catalog/src/main/assets/open_choice_with_validation_questionnaire.json (hunk 7 above)
Comment (jingtang10) [reply catalog/src/main/assets/open_choice_with_validation_questionnaire.json:36]: this is not done?

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDialogMultiSelectViewHolderFactoryEspressoTest.kt (hunk 10 above)
Comment (PallaviGanorkar) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDialogMultiSelectViewHolderFactoryEspressoTest.kt:190]: Done

**Review by jingtang10 — APPROVED** [review]
looks good. thanks @PallaviGanorkar

**Pull request merged**